javascript/app-framework/src/settings/application.ts

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import {
+	atom,
+	useRecoilValue,
+	useRecoilState,
+	useSetRecoilState,
+	type SetterOrUpdater,
+} from 'recoil'
+import { useCallback } from 'react'
+
+const LOCAL_STORAGE_KEY = 'app-framework-application-settings'
+const DEFAULT = {}
+
+const storageEffect =
+	(key: string) => ({ setSelf, onSet }: { setSelf: any; onSet: any }) => {
+		// initialize from ls if present
+		try {
+			const saved = localStorage.getItem(key)
+			setSelf(saved ? JSON.parse(saved) : DEFAULT)
+		} catch (_e) {
+			// quietly ignore ls errors
```

Comment:
Should we console.error out errors or log them out in some way in case there's an issue that devs need to look at/